baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -84,6 +91,54 @@ def stop(self) -> None:
         self.should_stop = True
 
 
+class KombuBatchConsumerWorker(ConsumerMixin, PumpWorker):
+    def __init__(
+        self,
+        connection: kombu.Connection,
+        queues: Sequence[kombu.Queue],
+        work_queue: BatchedQueue,
+        serializer: Optional[KombuSerializer] = None,
+        **kwargs: Any,
+    ) -> None:
+        self.connection = connection
+        self.queues = queues
+        self.work_queue = work_queue
+        self.serializer = serializer
+        self.kwargs = kwargs
+
+    def get_consumers(self, Consumer: kombu.Consumer, channel: Channel) -> Sequence[kombu.Consumer]:
+        args = dict(queues=self.queues, on_message=self.work_queue.put, **self.kwargs)
+        if self.serializer:
+            args["accept"] = [self.serializer.name]
+        return [Consumer(**args)]
+
+    def on_iteration(self) -> None:
+        self.work_queue.flush()
+
+    def stop(self) -> None:
+        logger.debug("Closing KombuBatchConsumerWorker.")
+        # `should_stop` is an attribute of `ConsumerMixin`
+        self.should_stop = True
+
+        messages = self.work_queue.drain()
+
+        if messages:
+            logger.debug("Requeueing %i messages", len(messages))
+            message: kombu.Message
+            for message in messages:
+                if not message.acknowledged:
+                    message.requeue()
+                else:
+                    logger.warning(
+                        "%s %s %s %s %s",
+                        "KombuBatchConsumerWorker is stopping and requeuing messages in the",
+                        "unprocessed batch. A message in this unprocessed batch had already",
+                        "been acknowledged, so it is not being requeued. However, this indicates",
+                        "an inconsistency in the batch processing logic which should be",
+                        "investigated.",
+                    )
```

Comment:
What's going on here? I'm guessing it was complaining about concatenating the strings with +? It should be possible to concatenate them without this formatting stuff by just putting them next to eachother with no operators:


```suggestion
                    logger.warning(
                        "KombuBatchConsumerWorker is stopping and requeuing messages in the "
                        "unprocessed batch. A message in this unprocessed batch had already "
                        "been acknowledged, so it is not being requeued. However, this indicates "
                        "an inconsistency in the batch processing logic which should be "
                        "investigated."
                    )
```